zulip_botserver/zulip_botserver/server.py

```diff
@@ -179,7 +207,13 @@ def main() -> None:
     options = parse_args()
     global bots_config
     try:
-        bots_config = read_config_file(options.config_file, options.bot_name)
+        if options.use_env_vars:
+            bots_config = read_config_from_env_vars(options.bot_name)
+        elif options.config_file:
+            bots_config = read_config_file(options.config_file, options.bot_name)
+        else:
+            sys.exit("""Configuration Error: Please pass --use-env-vars or -e along with properly formatted environment varibales
+                     Or pass the path to a properly formatted config file with --config-file path/to/config/file""")
     except MissingSectionHeaderError:
         sys.exit("Error: Your Botserver config file `{0}` contains an empty section header!\n"
                  "You need to write the names of the bots you want to run in the "
```

Comment:
It occurs to me on re-read hat this `except` (and the one below it) are specific to the `read_config_file` branch only; we can't raise them from `read_config_from_env_vars`.  We should move the `try .. except` from _outside_ the `if` to _inside` the `elif options.config_file:` block.